File: Printpress.UI/Printpress_Angular/src/app/features/order/models/item/item-upsert.Dto.ts
import { IObjectState } from "../../../../core/models/i-object-state";
import { ItemDetailsGetDto } from "../item-details/item-details-get.dto";


export interface ItemUpsertDto extends IObjectState {
    id: string;
    name: string;
    quantity: number;
    price: number;
    details: ItemDetailsGetDto[];
}

File: Printpress.Application/DTO/Order/Upsert/ItemUpsertDTO.cs
namespace Printpress.Application
{
    public class ItemUpsertDTO : TrackedDTO
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Quantity { get; set; }
        public decimal Price { get; set; }

        public List<ItemDetailsUpsertDTO> Details { get; set; }
    }
}
